cpp/harvest/marc_issn_lookup.cc

```diff
@@ -320,14 +372,40 @@ void ISSNLookup(char **argv, std::vector<CacheEntry> &journal_cache) {
                     // data is found
                     for (const auto &elemt : journal_cache) {
                         bool is_in_l = (std::find(elemt.issns_.begin(), elemt.issns_.end(), issn) != elemt.issns_.end() ? true : false);
-                        if (((elemt.preferred_issn_ == issn) || is_in_l) && elemt.is_valid_) {
+                        if ((elemt.preferred_issn_ == issn) || is_in_l) {
+                            if (elemt.is_valid_) {
+                                MARC::Subfields subfields(field.getSubfields());
+                                UpdateSubfieldUsingK10(subfields, elemt);
+                                field.setSubfields(subfields);
+                            }
+                            // issn is found in k10, ignoring wheather it is if valid or not
+                            is_issn_in_k10plus = true;
+                        }
+                    }
+                    if (not is_issn_in_k10plus) {
+                        IssnLookup::ISSNInfo issn_info;
+                        if (IsInISSNInfoCache(issn, *issn_org_cache, &issn_info)) {
+                            // issn is in the issn info cache already
                             MARC::Subfields subfields(field.getSubfields());
-                            UpdateSubfield(subfields, elemt);
+                            UpdateSubfieldUsingISSNOrg(subfields, issn_info);
                             field.setSubfields(subfields);
+                        } else {
+                            if (IssnLookup::GetISSNInfo(issn, &issn_info)) {
```

Comment:
If the ISSN is not in the cache, we should check whether there has been a failed lookup before and avoid doing multiple failed lookups for the same ISSN in that case (e.g. by checking ppns_with_issn_not_recognized before?)